pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -868,9 +868,13 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
 
     // server returns STRING as default dataType for all columns in (some) scenarios where no rows are returned
     // this is an attempt to return more faithful information based on other sources
+    // DISABLED until we face this issue:
+    // https://github.com/apache/pinot/issues/15064
+    /*
```

Comment:
Maybe we could conditionally disable this based on either a broker config or a query option so that environments that aren't affected by the issue with large IN clauses can continue using this functionality?

Reply:
@gortiz  asked me to not mess with this PR, although from what I see CI/CD is ignoring the `enabled=false` during the integration tests...

Anyway, there's a better PR that disables only the "use MSQE" strategy and allows to re-enable it using a config parameter:

https://github.com/apache/pinot/pull/15078